### ISSUE TITLE
Disable marking of violation when in Zen Mode

### ISSUE THREAD

**Issue by CGMossa** [DavidAnson/vscode-markdownlint#93]
Great extension and very much appreciated. It would be great, if the markings in editor would be disabled once the editor is in Zen Mode. Or atleast an option to enable/disable markdownlint in zenmode would be very much appreciated.

**Comment by DavidAnson** [comment]
It’ll depend on whether VS Code gives extensions a way to detect Zen Mode. I don’t see anything at https://code.visualstudio.com/api/references/vscode-api, but I’ll check again when I’m not on my phone.

**Comment by CGMossa** [comment]
Thanks. At this point, I just want to be able to disable it, as it is extremely distracting, but I don't want to disable it via the Extension menu, as then I may never remember to turn in it on again.

**Comment by wdscxsj** [comment]
@DavidAnson Is it possible to add a command to toggle it manually? In a quick read of other people's markdown files, one often has no need to do the fixes, and the markings often feel distracting.

**Comment by DavidAnson** [comment]
Yes, that seems straightforward. I expect the command would be a toggle for on/off and would take effect on the current file only and would not be saved. (For anything more permanent, the existing support for “ignore” should be used.)

**Comment by wdscxsj** [comment]
Or allow the user to choose whether or not it's a permanent choice? Suppose one has to skim over dozens of markdown files at the same time, he surely hates to turn it off one by one.

**Comment by DavidAnson** [comment]
Makes sense, yes. I could make it apply to the entire session. I just do not want it to persist because I worry people will forget they have change this and will be unable to figure out why the extension “isn’t working“. As long as they can close and reopen Code and things work again, that seems good.

**Comment by wdscxsj** [comment]
Thanks! I think that'll be very practical.

**Comment by DavidAnson** [comment]
@wdscxsj: The markdownlint.toggleLinting command is available in the `0.35.*` release to temporarily turn linting on/off. It can be run directly and/or bound to a keyboard shortcut.

@CGMossa: I looked into detecting Zen mode again and still don't see a way, so I'm leaving this issue open.

**Comment by wdscxsj** [comment]
@DavidAnson Confirmed. Thank you very much!

**Comment by DavidAnson** [comment]
Link to open issue in VS Code: https://github.com/microsoft/vscode/issues/114741
